tests/unit/components/Navigator/QuickNavigationModal.spec.js

```diff
@@ -9,73 +9,301 @@
+    ).toBe(filteredSymbols[0].title);
+    expect(
+      matchTitlesWrapper.at(1).text()
+      + matchTitlesWrapper.at(1).find(QuickNavigationHighlighter).props().text
+      + matchTitlesWrapper.at(1).find('span').text(),
+    ).toBe(filteredSymbols[1].title);
+  });
+
+  it('it redirects to the symbol path on symbol-match selection', async () => {
+    wrapper.setData({
+      debouncedInput: inputValue,
+    });
+    const referencesWrapper = wrapper.findAll(Reference);
+    expect(referencesWrapper.at(0).props().url).toBe(filteredSymbols[0].path);
+    expect(referencesWrapper.at(1).props().url).toBe(filteredSymbols[1].path);
+  });
+
+  it('it highlights the matching substring of the symbol title', async () => {
+    wrapper.setData({
+      debouncedInput: inputValue,
+    });
+    const matchTitlesWrapper = wrapper.findAll('.symbol-title');
+    expect(
+      matchTitlesWrapper.at(0).find(QuickNavigationHighlighter).props().matcherText,
+    ).toBe(symbolsMatchBlueprint[0].input);
+    expect(
+      matchTitlesWrapper.at(0).find(QuickNavigationHighlighter).props().text,
+    ).toBe(symbolsMatchBlueprint[0].subMatchString);
+    expect(
+      matchTitlesWrapper.at(1).find(QuickNavigationHighlighter).props().matcherText,
+    ).toBe(symbolsMatchBlueprint[1].input);
+    expect(
+      matchTitlesWrapper.at(1).find(QuickNavigationHighlighter).props().text,
+    ).toBe(symbolsMatchBlueprint[1].subMatchString);
+    expect(
+      matchTitlesWrapper.at(2).find(QuickNavigationHighlighter).props().matcherText,
+    ).toBe(symbolsMatchBlueprint[3].input);
+    expect(
+      matchTitlesWrapper.at(2).find(QuickNavigationHighlighter).props().text,
+    ).toBe(symbolsMatchBlueprint[3].subMatchString);
+  });
+
+  it('it debounces user input before filtering the symbols', () => {
+    wrapper.setData({
+      debouncedInput: inputValue,
+    });
+    expect(wrapper.vm.debouncedInput).toBe(inputValue);
+    wrapper.setData({
+      userInput: nonResultsInputValue,
+    });
+    expect(wrapper.vm.debouncedInput).toBe(inputValue);
+  });
+
+  it('it triggers new filtering on every debounce input change', () => {
+    const fuzzyMatch = jest.spyOn(wrapper.vm, 'fuzzyMatch');
+    wrapper.setData({
+      debouncedInput: inputValue,
+    });
+    wrapper.setData({
+      debouncedInput: nonResultsInputValue,
+    });
+    wrapper.setData({
+      debouncedInput: inputValue,
+    });
+    wrapper.setData({
+      debouncedInput: nonResultsInputValue,
+    });
+    expect(fuzzyMatch).toHaveBeenCalledTimes(4);
+  });
+
+  it('it matches the smallest matching substring from a symbol title', () => {
+    const customSymbols = [
+      {
+        title: 'foofooxyzbarbar',
+        path: '',
+        type: '',
+      },
+    ];
+    wrapper = shallowMount(QuickNavigationModal, {
+      propsData: {
+        children: customSymbols,
+        showQuickNavigationModal: true,
+      },
+      provide: {
+        quickNavigationStore: {},
```

Comment:
We should delete this one, right?

Reply:
yess e39a36f46e04ffc43c726b27d5dda242e891dc21